struct.md

```diff
@@ -0,0 +1,439 @@
+As of now it does not state how exporters are affected.
+
+## Proposed VSS 4.0 acceptance criteria and increments
+
+We need to decide how far we must go before we can release VSS 4.0, we does not need to go the whole way in one step
+
+### Increment 1
+
+* Syntax can be used without VSS-tools complaining
+* No semantic check by VSS-tools
+* No documentation
+* Not used in VSS standard catalog
+* Struct not supported by VSS-tools exporters
+
+
+### Increment 2
+
+* Semantic check on type-references by VSS-tools
+* Syntax documented
+* Well defined behavior for all exporters (e.g. either support, and/or give warning if used)
+
+### Increment 3
+
+* All "standard" exporters in vss-tools support structs
+* Guidelines on when to consider using structs rather than branches documented
+
+### Increment 4
+
+* We may start to use structs in VSS standard catalog.
+
+## Simple Usage
+
+```
+DeliveryInfo:
+  type: struct
+  description: A struct type containing info for each delivery
+  
+DeliveryInfo.Address:
+  datatype: string
+  type: item
+  description: Destination address
+
+DeliveryInfo.Receiver:
+  datatype: string
+  type: item
+  description: Name of receiver
+
+Delivery:
+  datatype: DeliveryInfo
+  type: sensor
+  description: Delivery
+```
+
+For VSS 4.0 it is not necessary that vss-tools do semantic check, i.e. if someone would add an extra `f` by mistake like this:
+
+```
+
+Delivery:
+  datatype: DeliveryInffo
+  comment: Note: Spelling error on line above, will only be detected if semantic check is implemented
+  type: sensor
+  description: Delivery
+```
+
+... then VSS-tools does not necessarily need to give an error (stretch goal to have semantic check that referred type exist).
+
+## Name resolution
+
+For now, two ways of referring to a type shall be considered correct:
+
+* Reference by (leaf) name to a struct definition within same branch
+* Reference by absolute path
+
+Relative paths (e.g. `../Powertrain.SomeStruct`) shall not be supported.
+Structs in parent branches will not be visible, in those cases absolute path needs to be used instead
+
+Examples:
+
+
+```
+A:
+  type: branch
+  description: Branch A.
+
+A.DeliveryInfo:
+  type: struct
+  
+A.DeliveryInfo.Address:
+  datatype: string
+  type: item
+
+A.DeliveryInfo.Receiver:
+  datatype: string
+  type: item
+
+A.Delivery1:
+  datatype: DeliveryInfo /* OK - As DeliveryInfo defined in same branch as Delivery1 */
+  type: sensor
+
+
+A.Delivery2:
+  datatype: A.DeliveryInfo /* OK - Addressing using absolute path */
+  type: sensor
+
+A.B:
+  type: branch
+
+A.B.Delivery3:
+  datatype: DeliveryInfo /* ERROR - No DeliverInfo defined in branch A.B */
+  type: sensor
+
+A.B.Delivery4:
+  datatype: A.DeliveryInfo /* OK - Addressing using absolute path */
+  type: sensor
+  
+A.B.Deliver5:
+  datatype: ../DeliveryInfo /* ERROR - Relative paths not supported */
+  type: sensor
+
+```
+
+### Order of declaration/definition
+
+The struct type must be defined before it is used.
```

Comment:
This requirement makes it impossible to reuse the struct definition in parts of the tree that lies before the definition.

Reply:
Meeting discussion:

- Ulf - should we have a separate branch on "top" like "Vehicle.DataDefinition.*" and require all struct definitions to be there? Makes tree cleaner, could obstruct the view.
- Krishna - having two passes is easy, order shall not matter in tree structure. Having dedicated branch can give problem, current proposal allows to reuse name. Having a separate top branch then we might need to replicate tree structure. 
- Erik: Do you want the use of a common top data folder to be a VSS style guide or something specified in syntax?
- Ulf: In syntax, reserved word
- Ulf/Krishna: use separate file for structs
- Erik: That means inline structs cannot be supported.
- Erik: What prefix to use for types
- Adnan: We need to discuss limitations so that people does not miss-use structs

Proposed way forward:

- Erik to adapt struct.md based on decision that structs shall be defined in a separate file and that tooling shall get two params (vspec struct file and vspec signal file)
- Erik (also) to propose/prepare "soft" limitations/recommendation, i.e. recommendations on when to use struct and when we from VSS will accept structs being added to VSS